integration-test-chronicle/src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/chronicle/ITVerifyChronicleBackend.java

```diff
@@ -0,0 +1,231 @@
+            "ALTER ROLE " + SUITE_SUPER_USER + " WITH OPTIONS = { 'grant_audit_whitelist_for_modify' : 'data'}"));
+        }
+
+        superCluster = cdt.createCluster(SUITE_SUPER_USER, "secret");
+        superSession = superCluster.connect();
+
+        testUsername = givenUniqueSuperuserWithMinimalWhitelist();
+        testCluster = cdt.createCluster(testUsername, "secret");
+        testSession = testCluster.connect();
+
+        ToolOptions options = ToolOptions
+                              .builder()
+                              .withPath(CassandraDaemonForAuditTest.getInstance().getAuditDirectory())
+                              .build();
+        reader = new QueueReader(options);
+    }
+
+    @Before
+    public void before()
+    {
+        // Drain the log
+        getRecords();
+    }
+
+    @After
+    public void after()
+    {
+        resetTestUserWithMinimalWhitelist(testUsername);
+    }
+
+    @AfterClass
+    public static void afterClass()
+    {
+        testSession.close();
+        testCluster.close();
+
+        for (int i = 0; i < usernameNumber.get(); i++)
+        {
+            superSession.execute(new SimpleStatement("DROP ROLE IF EXISTS " + SUITE_TEST_USER_PREFIX + i));
+        }
+        superSession.close();
+        superCluster.close();
+
+        try (Cluster cassandraCluster = cdt.createCluster();
+             Session cassandraSession = cassandraCluster.connect())
+        {
+            cassandraSession.execute(new SimpleStatement("DROP ROLE IF EXISTS " + SUITE_SUPER_USER));
+        }
+    }
+
+    @Test
+    public void simpleUpdateIsLogged()
+    {
+        givenTable("ks1", "tbl");
+        String username = givenSuperuserWithMinimalWhitelist();
+
+        testSession.execute("UPDATE ks1.tbl SET value = 'hepp' WHERE key = 88");
+
+        thenChronicleLogContainEntryForUser("UPDATE ks1.tbl SET value = 'hepp' WHERE key = 88", username);
+    }
+
+    @Test
+    public void preparedInsertIsLogged()
+    {
+        givenTable("ks2", "tbl");
+        String username = givenSuperuserWithMinimalWhitelist();
+
+        PreparedStatement preparedStatement = testSession.prepare("INSERT INTO ks2.tbl (key, value) VALUES (?, ?)");
+        testSession.execute(preparedStatement.bind(5, "hepp"));
+
+        thenChronicleLogContainEntryForUser("INSERT INTO ks2.tbl (key, value) VALUES (?, ?)[5, 'hepp']", username);
+    }
+
+    private void thenChronicleLogContainEntryForUser(String operation, String username)
+    {
+        List<AuditRecord> records = getRecords();
+        assertThat(records).hasSize(1);
+
+        AuditRecord record = records.get(0);
+        assertThat(record.getOperation()).isEqualTo(operation);
+        assertThat(record.getUser()).isEqualTo(username);
+        assertThat(record.getStatus()).isEqualTo("ATTEMPT");
+        assertThat(record.getClient()).isEqualTo(InetAddress.getLoopbackAddress());
+        assertThat(record.getTimestamp()).isLessThan(System.currentTimeMillis());
```

Comment:
If only "isLessThan" is checked, then getTimestamp() could always return 0 and this test would still succeed... Maybe we could add an isGreaterThan check as well? Say currentTime - 30 sec ?